airgen/src/lib.rs

```diff
@@ -174,12 +174,26 @@ impl<'a, T: FieldElement> ASMPILConverter<'a, T> {
         LinkDefinitionStatement {
             source: _,
             flag,
-            params,
-            to: CallableRef { instance, callable },
+            params: lhs_params,
+            to:
+                CallableRef {
+                    instance,
+                    callable,
+                    params: rhs_params,
+                },
         }: LinkDefinitionStatement<T>,
     ) -> Link<T> {
+        // The actual mapping of parameters used for the link constraint comes
+        // from the RHS of the instr/link declaration (as it allows for extra
+        // registers/columns being used and parameter reordering).
+        // We allow declarations with an empty RHS as syntactic sugar for when RHS = LHS.
+        let params = if !rhs_params.is_empty() {
+            rhs_params
+        } else {
+            lhs_params
+        };
```

Comment:
This might be nitpicking, but if there is no compelling reason to do otherwise, I often try to avoid negations and instead switch the both cases, because statements without negations are usually easier to understand:
```suggestion
        let params = if rhs_params.is_empty() {
            lhs_params
        } else {
            rhs_params
        };
```